lldb/packages/Python/lldbsuite/test/lldbplatformutil.py

```diff
@@ -113,6 +130,10 @@ def _get_platform_os(p):
             platform = "openbsd"
         return platform
 
+    # Triple is not available if we're not connected yet
+    if p.GetName() == "remote-linux":
+        return "linux"
+
```

Comment:
Can you explain why was this necessary? I get the connected part, but who was asking for the platform before connecting?

Reply:
That may be relevant if _get_platform_os is called before the connection is established, or, for example, if it has failed to be established.
But yeah, currently I'm not aware of such use cases. Should it be removed?

Reply:
Yes, I would very much prefer that, as the code you added only handles linux. To support other oses, we would need to create a whole new mapping from lldb platform names to os strings. That list could then get out of sync, etc., so it would be best to just avoid doing that.

I'm not worried about the debug target going down during testing, as that means the test results will be meaningless anyway. If we find out we regularly need the os name before connecting, then I think we ought to revisit the mechanism for determining the OS.

Reply:
Fixed.